### PR TITLE
Day07

### PR DESCRIPTION
**Pull request by davidabrahams**

### CODE REVIEW

**Comment by thecardkid** [comment]
🚢

File: day07/src/WikiNodeExample.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        String url = "https://en.wikipedia.org/wiki/Java_(programming_language)";
+
+        // download and parse the document
+        Connection conn = Jsoup.connect(url);
+        Document doc = conn.get();
+
+        // select the content text and pull out the paragraphs.
+        Element content = doc.getElementById("mw-content-text");
+
+        // TODO: avoid selecting paragraphs from sidebars and boxouts
```
Comment (ssingal05) [inline day07/src/WikiNodeExample.java]: Remove this TODO if its Allen's

File: day07/src/WikiFetcher.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        sleepIfNeeded();
+
+        // download and parse the document
+        Connection conn = Jsoup.connect(url);
+        Document doc = conn.get();
+
+        // select the content text and pull out the paragraphs.
+        Element content = doc.getElementById("mw-content-text");
+
+        // TODO: avoid selecting paragraphs from sidebars and boxouts
```
Comment (ssingal05) [inline day07/src/WikiFetcher.java]: See above

File: day07/src/WikiFetcher.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        // assemble the file name
+        String slash = File.separator;
+        String filename = "resources" + slash + realURL.getHost() + realURL.getPath();
+
+        // read the file
+        InputStream stream = WikiFetcher.class.getClassLoader().getResourceAsStream(filename);
+        Document doc = Jsoup.parse(stream, "UTF-8", filename);
+
+        // parse the contents of the file
+        // TODO: factor out the following repeated code
```
Comment (ssingal05) [inline day07/src/WikiFetcher.java]: Allen TODO

**Pull request merged**